project/source/Scripts/Objects/LabObject.gd

```diff
@@ -11,32 +11,35 @@ class_name LabObject
 var tooltip: Label #Set when it's created
 
 #these variables are used internally to handle clicking and dragging:
-var dragging = false
-var dragOffset = Vector2(0, 0)
-var dragStartGlobalPos = null
-var maxMovementForActIndependently = 10 #If an object is released more than this distance fron where it started being dragged (line above), it will only interact, not act independently.
-@onready var defaultMode = mode
-@onready var defaultZIndex = z_index
-@onready var defaultZAsRelative = z_as_relative
+var dragging:bool = false
+var dragOffset:Vector2 = Vector2(0, 0)
+var dragStartGlobalPos:Vector2 = Vector2()
+var maxMovementForActIndependently:int = 10 #If an object is released more than this distance fron where it started being dragged (line above), it will only interact, not act independently.
+#@onready var defaultMode = mode
+#temporarily disabling the deafaultMode var since this changed in gd4
+# trying different workarounds currently
```

Comment:
It looks like `mode` was replaced with a combination of `freeze`, `freeze_mode`, and `lock_rotation` (plus maybe some other things). But the only place it seems to be touched is when a `LabObject` is dragged, during which time it is set to be a kinematic object; this can be done just by setting `freeze = true` and `freeze_mode = FREEZE_MODE_KINEMATIC`, so we probably only need to bother saving and restoring those two variables.